components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5336,14 +5336,28 @@ public void updatePermissionListOfGroup(String groupName, String[] permissionToA
 
     /**
      * Returns all attributes belong to the extension schema dialect.
+     * unless its mapped local dialect claim value "support by default value is false".
      *
      * @param scimClaimToLocalClaimMap
      * @return Map of attribute.
      */
     private Map<String, Attribute> getAllScimSchemaAttributes(Map<ExternalClaim, LocalClaim> scimClaimToLocalClaimMap) {
 
-        return scimClaimToLocalClaimMap.entrySet().stream().map(e -> getSchemaAttributes(e.getKey(), e.getValue(),
-                true)).collect(Collectors.toMap(attr -> attr.getName(), Function.identity()));
+        Map<String, Attribute> filteredFlatAttributeMap = new HashMap<>();
+
+        for (Map.Entry<ExternalClaim, LocalClaim> entry : scimClaimToLocalClaimMap.entrySet()) {
```

Comment:
we can reuse getFilteredEnterpriseUserSchemaAttributes() method. We can rename that private method and use for this custom schema as well,

Remove this getAllScimSchemaAttributes since it is filtering the claims, not returning all claims

Reply:
Seems like we can use the `getFilteredUserSchemaAttributes` . It filter for userame claim

Reply:
Fixed in commit 4c020f167c3eab9cd4d13fcf6cfa9c8a2426300c